zookeeper-server/src/main/java/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -112,14 +112,17 @@ public String getLoginContext() {
         return null;
     }
 
-    public ZooKeeperSaslClient(final String serverPrincipal, ZKClientConfig clientConfig) throws LoginException {
+    public ZooKeeperSaslClient(ZKClientConfig clientConfig) throws LoginException {
+        this.clientConfig = clientConfig;
+    }
+
+    public void createSaslClient(final String serverPrincipal) throws LoginException {
```

Comment:
We could keep the serverPrincipal into a field, this way we preserve the original behaviour, that this value is set during the construction.
Otherwise this method may be called with the wrong value.
Does it make sense to you?

Reply:
@eolivelli , Makes sense. Update the files in the review.